crypto/jwk.go

```diff
@@ -94,6 +112,24 @@ func PublicKeyToPublicKeyJWK(key crypto.PublicKey) (*PublicKeyJWK, error) {
 	}
 }
 
+// PrivateKeyToJWK converts a private key to a JWK
+func PrivateKeyToJWK(key crypto.PrivateKey) (jwk.Key, error) {
+	switch key.(type) {
+	case rsa.PrivateKey:
+		return jwkKeyFromRSAPrivateKey(key.(rsa.PrivateKey))
+	case ed25519.PrivateKey:
+		return jwkKeyFromEd25519PrivateKey(key.(ed25519.PrivateKey))
+	case x25519.PrivateKey:
+		return jwkKeyFromX25519PrivateKey(key.(x25519.PrivateKey))
+	case secp256k1.PrivateKey:
+		return jwkKeyFromSECP256k1PrivateKey(key.(secp256k1.PrivateKey))
+	case ecdsa.PrivateKey:
+		return jwkKeyFromECDSAPrivateKey(key.(ecdsa.PrivateKey))
```

Comment:
```suggestion
	switch k := key.(type) {
	case rsa.PrivateKey:
		return jwkKeyFromRSAPrivateKey(k)
	case ed25519.PrivateKey:
		return jwkKeyFromEd25519PrivateKey(k)
	case x25519.PrivateKey:
		return jwkKeyFromX25519PrivateKey(k)
	case secp256k1.PrivateKey:
		return jwkKeyFromSECP256k1PrivateKey(k)
	case ecdsa.PrivateKey:
		return jwkKeyFromECDSAPrivateKey(k)
```

Reply:
good idea, let me update this in the other one too